cmd/client/resources/member.go

```diff
@@ -0,0 +1,222 @@
+		if err != nil {
+			general.ExitWithErrorf("unmarshal options failed: %v", err)
+		}
+		for k, v := range optionMap {
+			result[k] = v
+		}
+		return result
+	}
+
+	results := []map[string]interface{}{}
+	for _, status := range memberStatus {
+		results = append(results, statusToMapInterface(status))
+	}
+	general.PrintMapInterface(results, []string{
+		"name", "lastHeartbeatTime", "", "etcd", "",
+	}, []string{})
+}
+
+func DeleteMember(cmd *cobra.Command, names []string) error {
+	for _, name := range names {
+		_, err := handleReq(http.MethodDelete, makePath(general.MemberItemURL, name), nil)
+		if err != nil {
+			return general.ErrorMsg("purge", err, fmt.Sprintf("%s %s", MemberKind, name))
+		}
+		fmt.Println(general.SuccessMsg("purge", MemberKind, name))
+	}
+	return nil
+}
+
+func GetMember(cmd *cobra.Command, args *general.ArgInfo) (err error) {
+	msg := "all " + MemberKind
+	if args.ContainName() {
+		msg = fmt.Sprintf("%s %s", MemberKind, args.Name)
+	}
+	getErr := func(err error) error {
+		return general.ErrorMsg(general.GetCmd, err, msg)
+	}
+
+	body, err := handleReq(http.MethodGet, makePath(general.MembersURL), nil)
+	if err != nil {
+		return getErr(err)
+	}
+
+	if !general.CmdGlobalFlags.DefaultFormat() {
+		general.PrintBody(body)
+		return nil
+	}
+
+	members := []*cluster.MemberStatus{}
+	err = codectool.Unmarshal(body, &members)
+	if err != nil {
+		return getErr(err)
+	}
+	if !args.ContainName() {
+		printMemberStatusTable(members)
+		return nil
+	}
+
+	for _, member := range members {
+		if member.Options.Name == args.Name {
+			printMemberStatusTable([]*cluster.MemberStatus{member})
+			return nil
+		}
+	}
+	return getErr(fmt.Errorf("member %s not found", args.Name))
+}
+
+type member struct {
+	Name      string
+	Role      string
+	APIAddr   string
+	Heartbeat *time.Time
+
+	State     string
+	StartTime *time.Time
+}
+
+func (m *member) age() string {
+	if m.StartTime == nil {
+		return "unknown"
+	}
+	return general.DurationMostSignificantUnit(time.Since(*m.StartTime))
+}
+
+func (m *member) heartbeatAge() string {
+	if m.Heartbeat == nil {
+		return "unknown"
+	}
+	return general.DurationMostSignificantUnit(time.Since(*m.Heartbeat)) + " ago"
+}
+
+func newMember(ms *cluster.MemberStatus) *member {
+	member := &member{}
+	member.Name = ms.Options.Name
+	member.Role = ms.Options.ClusterRole
+	member.APIAddr = ms.Options.APIAddr
+
+	heartbeat, err := time.Parse(time.RFC3339, ms.LastHeartbeatTime)
+	if err == nil {
+		member.Heartbeat = &heartbeat
+	}
+
+	if ms.Etcd != nil {
+		startTime, err := time.Parse(time.RFC3339, ms.Etcd.StartTime)
+		if err == nil {
+			member.StartTime = &startTime
+		}
+		member.State = ms.Etcd.State
+	}
+	return member
+}
+
+func printMemberStatusTable(memberStatus []*cluster.MemberStatus) {
+	// Output table:
+	// NAME  ROLE  AGE  STATE  API-ADDR  HEARTBEAT
+	table := [][]string{}
+	table = append(table, []string{"NAME", "ROLE", "AGE", "STATE", "API-ADDR", "HEARTBEAT"})
```

Comment:
Thanks for your advice. I check several k8s commands, I find some of them use seperate words(pod), some of them use "-" to connect words(node, deployment, svc). I am prefer "API-ADDR" over "API ADDR" because "API-ADDR" feel more like one word, and "API ADDR" feel more like two words. And when print table in terminal, sometimes two columns really close to each other, "API-ADDR" may feels better. So, it that ok to keep "API-ADDR"?
```
cd:➜  ~  |>k get svc -o wide | grep NAME                                                                                                                                                         [~]
NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                     AGE    SELECTOR
cd:➜  ~  |>k get po -o wide | grep NAME                                                                                                                                                          [~]
NAME                                                 READY   STATUS      RESTARTS       AGE     IP              NODE   NOMINATED NODE   READINESS GATES
cd:➜  ~  |>k get deploy -o wide | grep NAME                                                                                                                                                      [~]
NAME                                READY   UP-TO-DATE   AVAILABLE   AGE    CONTAINERS                          IMAGES                                                                                     SELECTOR
cd:➜  ~  |>k get node -o wide | grep NAME                                                                                                                                                        [~]
NAME   STATUS   ROLES                  AGE    VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION   CONTAINER-RUNTIME
```